wakuv2/waku.go

```diff
@@ -218,13 +219,28 @@ func New(nodeKey string, fleet string, cfg *Config, logger *zap.Logger, appDB *s
 	var privateKey *ecdsa.PrivateKey
 	if nodeKey != "" {
 		privateKey, err = crypto.HexToECDSA(nodeKey)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert nodekey into a valid private key: %v", err)
+		}
 	} else {
-		// If no nodekey is provided, create an ephemeral key
-		privateKey, err = crypto.GenerateKey()
-	}
+		nodeKeyStr := os.Getenv("WAKUV2_NODE_KEY")
```

Comment:
Why not do this outside, before calling `New`, and pass the key to the `New` function?

Reply:
> @stefandunca : Why not do this outside, before calling New, and pass the key to the New function as parameter?

I put it in the New function for no reason in particular. I'll move this out to outside as suggested.